e2e/cypress/integration/messaging/MM-T134-visual-verification-of-tooltips_spec.js

```diff
@@ -0,0 +1,103 @@
+            testTeam = team;
+            loggedUser = user;
+
+            cy.apiAdminLogin;
+
+            cy.apiAddUserToTeam(testTeam.id, loggedUser.id);
+
+            cy.visit(`/${testTeam.name}/channels/town-square`);
+        });
+    });
+    it('MM-T134-visual-verification-of-tooltips', () => {
+        // * Hover effect wraps around Date and New Messages lines
+        // supposed to validate shadow over effect wraping around date but not sure if this is possible
+        cy.get('#postListContent').find('.top').should('be.visible');
+
+        // * Members tool-tip is present
+        cy.findAllByLabelText('members').should('be.visible').trigger('mouseover');
+        cy.get('div.tooltip-inner').should('be.visible').and('contain', 'Members');
+        cy.findAllByLabelText('members').should('be', 'visible').trigger('mouseout');
+
+        // * Pinned post tool-tip is present
+        cy.findAllByLabelText('Pinned posts').should('be.visible').trigger('mouseover');
+        cy.get('div.tooltip-inner').should('be.visible').and('contain', 'Pinned posts');
+        cy.findAllByLabelText('Pinned posts').should('be', 'visible').trigger('mouseout');
+
+        // * Saved posts tool-tip is present
+        cy.findAllByLabelText('Saved posts').should('be.visible').trigger('mouseover');
+        cy.get('div.tooltip-inner').should('be.visible').and('contain', 'Saved posts');
+        cy.findAllByLabelText('Saved posts').should('be', 'visible').trigger('mouseout');
+
+        // * Add to favourites posts tool-tip is present - un checked
+        cy.findAllByLabelText('add to favorites').should('be.visible').trigger('mouseover');
+        cy.get('div.tooltip-inner').should('be.visible').and('contain', 'Add to Favorites');
+        cy.findAllByLabelText('add to favorites').should('be', 'visible').trigger('mouseout');
+
+        // * Add to favourites posts tool-tip is present - checked
+        cy.findAllByLabelText('add to favorites').click();
+        cy.findAllByLabelText('remove from favorites').should('be.visible').trigger('mouseover');
+        cy.get('div.tooltip-inner').should('be.visible').and('contain', 'Remove from Favorites');
+        cy.findAllByLabelText('remove from favorites').should('be', 'visible').trigger('mouseout');
+
+        // * Unmute a channel tool-tip is present
+        cy.findAllByLabelText('dropdown icon').click();
+        cy.findAllByText('Mute Channel').click();
+        cy.findAllByLabelText('Muted Icon').should('be', 'visible').trigger('mouseover');
+        cy.get('div.tooltip-inner').should('be.visible').and('contain', 'Unmute');
+        cy.findAllByLabelText('Muted Icon').should('be', 'visible').trigger('mouseout');
+
+        // * Long channel name (shown truncated on the LHS)
+        cy.apiCreateChannel(testTeam.id, 'channel-test', `Public channel with a long name-${timestamp}`).then((res) => {
+            verifyChannel(res);
+        });
+
```

Comment:
The test misses to validate the following points mentioned in the TM4J test:
- 'Download' on download file icon
- Hover effect wraps around Date and New Messages lines (no tooltip just proper post shading needs to be verified on hover)

Reply:
Both fixed.